grc/gui/Executor.py

```diff
@@ -56,7 +56,10 @@ def _popen(self):
         # When in no gui mode on linux, use a graphical terminal (looks nice)
         xterm_executable = find_executable(self.xterm_executable)
         if generator.generate_options == 'no_gui' and xterm_executable:
-            run_command_args = [xterm_executable, '-e', run_command]
+            if ('gnome-terminal' in xterm_executable):
+                run_command_args = [xterm_executable, '--'] + run_command_args
```

Comment:
Do you need run_command somewhere on this line?

Reply:
run_command is the non-splitted version of run_command_args.  When calling gnome-terminal with `--` it needs the splitted version due to where the quotation marks get put